app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -1644,18 +1653,23 @@ public ArtifactVersionMetaDataDto getArtifactVersionMetaData(String groupId, Str
 
     private ArtifactVersionMetaDataDto getArtifactVersionMetaDataInternal(String groupId, String artifactId, String version) {
         try {
-            return this.jdbi.withHandle( handle -> {
+            return this.handles.withHandle( handle -> {
                 String sql = sqlStatements.selectArtifactVersionMetaData();
-                return handle.createQuery(sql)
+                Optional<ArtifactVersionMetaDataDto> res = handle.createQuery(sql)
                         .bind(0, tenantContext.tenantId())
                         .bind(1, normalizeGroupId(groupId))
                         .bind(2, artifactId)
                         .bind(3, version)
                         .map(ArtifactVersionMetaDataDtoMapper.instance)
-                        .one();
+                        .findOne();
+                if (res.isEmpty()) {
```

Comment:
Just a preference of mine, but I usually prefer to check the positive case, with `ifPresent`. I find it more readable.

Reply:
Very good suggestion using `orElseThrow`.  I have changed all the places where this was relevant.